File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <bits/stdc++.h>
using namespace std;
// 가장 깔끔하고 정리된 방법
int solution(vector<int> nums) {
    unordered_set<int> s(nums.begin(), nums.end());

    return min(nums.size() / 2, s.size());
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;
class Solution {
    public int solution(int[] nums) {
        Set<String> mySet2 = new HashSet<>();
        for (int num : nums) {
            mySet2.add(String.valueOf(num));
        }
        int answer = Math.min(mySet2.size(), nums.length / 2);
        return answer;
    }
}